src/janus.c

```diff
@@ -3380,8 +3380,16 @@ void janus_transport_incoming_request(janus_transport *plugin, janus_transport_s
 	JANUS_LOG(LOG_VERB, "Got %s API request from %s (%p)\n", admin ? "an admin" : "a Janus", plugin->get_package(), transport);
 	/* Create a janus_request instance to handle the request */
 	janus_request *request = janus_request_new(plugin, transport, request_id, admin, message);
-	/* Enqueue the request, the thread will pick it up */
-	g_async_queue_push(requests, request);
+	if(message == NULL) {
+		/* Report details about the parsing error */
+		janus_process_error(request, /*session_id=*/0, /*transaction=*/NULL, JANUS_ERROR_INVALID_JSON,
```

Comment:
Please do let me know if there are any issues with reporting results from the websocket messaging thread for this error case. We could cache the error information instead, if necessary.

Reply:
I'd rather put the check (and response) where I added it in my PR. You're only taking into account the WebSocket transport since that's what you use, but we have many different transports, each with their own threading model, and more may be added in the future. Adding it where I added it would ensure consistency with how the core usually takes care of responses.

As a side note, please don't add those inline comments for session ID and transaction.

Reply:
Moved the handling into the request management thread.